dask/dataframe/core.py

```diff
@@ -6524,6 +6484,65 @@ def func(partition_info, *args, **kwargs):
     return new_dd_object(graph, name, meta, divisions)
 
 
+def _get_divisions(align_dataframes, transform_divisions, dfs, func, args, kwargs):
+    if align_dataframes:
+        divisions = dfs[0].divisions
+    else:
+        # Unaligned, dfs is a mix of 1 partition and 1+ partition dataframes,
+        # use longest divisions found
+        divisions = max((d.divisions for d in dfs), key=len)
+    if transform_divisions and isinstance(dfs[0], Index) and len(dfs) == 1:
+        try:
+            divisions = func(
+                *[pd.Index(a.divisions) if a is dfs[0] else a for a in args], **kwargs
+            )
+            if isinstance(divisions, pd.Index):
+                divisions = methods.tolist(divisions)
+        except Exception:
+            pass
+        else:
+            if not valid_divisions(divisions):
+                divisions = [None] * (dfs[0].npartitions + 1)
+    return divisions
+
+
+def _is_only_scalar(args):
+    return all(isinstance(arg, Scalar) for arg in args)
+
+
+def _get_meta(args, dfs, func, kwargs, meta, parent_meta):
```

Comment:
This helper seems very specific to `map_partitions` and `map_overlap`. Do you have thoughts on how to make it more generic? A first step could be a helpful docstring. If we don't want this to be general, maybe we can change the name to `_get_meta_map_partitions` or similar?

Reply:
Agreed, no idea to make it more generic for the moment, I'll rename it to _get_meta_map_partitions and add a docstring